seisbench/models/dpp.py

```diff
@@ -223,3 +526,525 @@ def forward(self, x):
+                    to_write.append(end_element[2])
+                    start_sorted.put(start_element)
+                else:
+                    active.put(
+                        (
+                            start_element[2].stats.endtime,
+                            start_element[1],
+                            start_element[2],
+                        )
+                    )
+                    active.put(end_element)
+
+            if not strict and active.qsize() == 0 and len(to_write) != 0:
+                t0 = min(trace.stats.starttime for trace in to_write)
+                t1 = max(trace.stats.endtime for trace in to_write)
+
+                data = np.zeros(
+                    (len(component_order), int((t1 - t0) * sampling_rate + 2))
+                )  # +2 avoids fractional errors
+
+                for trace in to_write:
+                    p = int((trace.stats.starttime - t0) * sampling_rate)
+                    cidx = comp_dict[trace.id[-1]]
+                    data[cidx, p : p + len(trace.data)] = trace.data
+
+                data = data[:, :-1]  # Remove fractional error +1
+
+                output_times.append(t0)
+                output_data.append(data)
+
+                to_write = []
+
+            if strict and active.qsize() == len(component_order):
+                traces = []
+                while not active.empty():
+                    traces.append(active.get()[2])
+
+                t0 = max(trace.stats.starttime for trace in traces)
+                t1 = min(trace.stats.endtime for trace in traces)
+
+                short_traces = [trace.slice(t0, t1) for trace in traces]
+                data = np.zeros(
+                    (len(component_order), len(short_traces[0].data) + 2)
+                )  # +2 avoids fractional errors
+                for trace in short_traces:
+                    cidx = comp_dict[trace.id[-1]]
+                    data[cidx, : len(trace.data)] = trace.data
+
+                data = data[:, :-2]  # Remove fractional error +2
+
+                output_times.append(t0)
+                output_data.append(data)
+
+                for trace in traces:
+                    if t1 < trace.stats.endtime:
+                        start_sorted.put((t1, seqnum, trace.slice(starttime=t1)))
+                        seqnum += 1
+
+        return output_times, output_data
+
+    def classify_aggregate(self, annotations, picks_in, argdict):
+        """
+        Converts the annotations to discrete picks using :py:func:`~seisbench.models.base.WaveformModel.picks_from_annotations`.
+        Trigger onset thresholds for picks are derived from the argdict at keys "[phase]_threshold".
+
+        :param annotations: See description in superclass
+        :param argdict: See description in superclass
+        :return: List of picks
+        """
+        picks = []
+        for annotation in annotations:
+            phase = annotation[0].stats.channel.split("_")[-1][0]
```

Comment:
Exactly, my idea was that those annotations (MCD samples) were returned as a list of streams by the overwritten `annotate` function. It was just the way I thought this could be implemented following the existent logic for detection.
But I can see this would be inconsistent with the functionality intended for other models, so I'm okay with your suggestions.
However, `classify_aggregate` would still need to receive as input annotations from `annotate` which should contain the MCD samples for predicted picks.